docs/src/manual/advanced.md

```diff
@@ -34,6 +34,31 @@ p.constraints[1].dual
 constraint.dual
 ```
 
+## Warmstarting
+
+If you're solving the same problem many times with different values of a
+parameter, Convex.jl can initialize many solvers with the solution to the
+previous problem, which sometimes speeds up the solution time. This is called a
+**warm start**.
+
+To use this feature, pass the optional argument `warmstart=true` to the `solve!`
+method.
+
+```julia
+using Convex, SCS
+n = 1_000
+y = rand(n)
+x = Variable(n)
+lambda = Variable(Positive())
+fix!(lambda, 100)
+problem = minimize(sumsquares(y - x) + lambda * sumsquares(x - 10))
+@time solve!(problem, SCS.Optimizer)
+# Now warmstart. If the solver takes advantage of warmstarts, this run will be
+# faster
+fix!(lambda, 105)
+@time solve!(problem, SCS.Optimizer; warmstart = true)
```

Comment:
This example doesn't really make sense because SCS requires a complete start, and Convex may introduce arbitrary numbers of intermediate variables.

Reply:
In MOI bridges, when we introduce intermediate variables, we set the starting values for them as well. Maybe that's another motivation to move these parts to use the custom MOI cones and let the bridges do the reformulation.

Reply:
I realized that the loop will account for these values after a solve. You just can't set their value _before_ a solve.

Reply:
I think the SCS MOI wrapper should throw a warning if the starting values are only partially set.
Is this a regression from the previous release of Convex.jl ?

Reply:
Ah. SCS uses `0` if default:
https://github.com/jump-dev/SCS.jl/blob/e362f9551ba9f977487263cb5162d5801bf5cb3b/src/MOI_wrapper/MOI_wrapper.jl#L345

Your confusion is because Convex conflates `VariablePrimalStart` with `VariablePrimal` If `warmstart = true`, it looks up `.value`, which is also where the primal solution is stored.

Reply:
So what about your first message in this thread ? Does the example make sense or not ?

Reply:
In retrospect it does make sense.